Semantics.md

```diff
@@ -0,0 +1,67 @@
+
+### Enumerations
+
+Don't use the [Flags] attribute, instead use a set of the enumeration. This adheres to the general principle that when all else is equal, prefer typed over untyped code.
+
+Bad:
+```csharp
+[Flags]
+enum Inventory
+{
+	Hat,
+	Sword,
+	BigSword
+}
+
+class Player
+{
+	Inventory Items
+	{
+		get;
+		set;
+	}
+
+	int CalculateDamage()
+	{
+		int result = 0;
+		if (Items.HasFlag(Inventory.Sword))
+			result += 1;
+		if (Items == Inventory.BigSword) //Was the omission of HasFlag intended?
+			result += 2;
+		return result;
+	}
+}
+```
+
+Good:
+```csharp
+enum Item
```

Comment:
Hi David,

Thank you for your feedback. I've shortened the examples. Are they short
enough, or did you have something else in mind? Furthermore I've made it
clear that [Flags] can be useful for performance reasons.

Cheers,
Rermy.

On Wed, Nov 4, 2015 at 6:09 PM, David Siegel <notifications@github.com>
wrote:

> In Semantics.md
> <https://github.com/dvdsgl/csharp-in-style/pull/12#discussion_r43908505>:
>
> > +
> > +	int CalculateDamage()
> > +	{
> > +		int result = 0;
> > +		if (Items.HasFlag(Inventory.Sword))
> > +			result += 1;
> > +		if (Items == Inventory.BigSword) //Was the omission of HasFlag intended?
> > +			result += 2;
> > +		return result;
> > +	}
> > +}
> > +```
> > +
> > +Good:
> > +```csharp
> > +enum Item
>
> One-liner, to keep sample brief.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dvdsgl/csharp-in-style/pull/12/files#r43908505>.
>